adhocteam/Head-Start-TTADP
# OWASP finding remediation

**Description of change**

Sets the `form-action` directive on the CSP header.

This also adds some scripts to make it easier to run owasp locally within docker.

**How to test**

1. Run `./bin/prod-style-server`
2. Run `./bin/run-owasp-scan head-start-ttadp_smarthub` (you may need to change the argument depending on the network docker assigned in step 1)
3. See that the `reports/owasp_report.html` doesn't include the finding about CSP `form-action`


**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/306
* https://github.com/HHS/Head-Start-TTADP/issues/95

**Checklist**

    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
